django/contrib/admin/utils.py

```diff
@@ -506,11 +526,19 @@ def construct_change_message(form, formsets, add):
                         }
                     })
                 for changed_object, changed_fields in formset.changed_objects:
+                    changed_fields_labels = []
```

Comment:
This added block of code is essentially the same as above. Can you pull it out into a separate function that takes the form and changed fields to avoid the duplication?